crates/bindings-csharp/BSATN.Codegen/Utils.cs

```diff
@@ -133,6 +133,8 @@ static string GetTypeInfoForNamedType(INamedTypeSymbol type)
                 // (U)Int128 are not treated by C# as regular primitives, so we need to match them by type name.
                 "System.Int128" => "SpacetimeDB.BSATN.I128",
                 "System.UInt128" => "SpacetimeDB.BSATN.U128",
+                "SpacetimeDB.Int256" => "SpacetimeDB.BSATN.I256",
```

Comment:
For custom types it's better to inline the `struct BSATN` implementation right inside the type instead of augmenting the codegen.

The only reason this codegen has a large switch is because C# doesn't have Rust-like traits that would allow us to declare BSATN implementations on built-in types - but where types are not built-in, we don't need to do this.

Reply:
In fact, you don't even need to provide one manually - just use lower-endian field order and then you can use `[SpacetimeDB.Type]` for automatic read/write generation on those types.

Reply:
I'd just really like to get rid of this hardcoded `switch` in the future - e.g. one idea I had was just defining a `SystemShims` namespace that would be used for any types from the `System` namespace - and adding non-`System` types kinda interferes with that.

I can remove it when I get to it, but I'd prefer not to have non-System types in here to begin with.